### PR TITLE
Restore Quote when clicking browser back button

### PR DESCRIPTION
**Pull request by Dinwy**
# Description

This pull request addresses critical issues and improvements for the Komoju Payments module in Magento, focusing on webhook error fixes, code cleanup, and enhanced payment flow.

## Key Changes

- Restore Quote
  - Added RestoreQuoteFromSession observer to restore the quote when user click browser back button

- Webhook Error Fixes:
  - Resolved issues causing webhook failures on ping.

- Code Cleanup:
  - Removed redundant code and improved readability.

## Videos

- Clicking the browser back button will restore quote(It was making the cart empty before)

https://github.com/degica/komoju-magento/assets/16696001/2ca13478-959d-47e9-a348-86e47d8f23af

### CODE REVIEW

File: src/app/code/Komoju/Payments/Observer/RestoreQuoteFromSession.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Session\SessionManager;
+use Magento\Sales\Model\Order;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+class RestoreQuoteFromSession implements ObserverInterface
```
Comment (rramsden) [inline src/app/code/Komoju/Payments/Observer/RestoreQuoteFromSession.php:24]: @Dinwy We should add some module documentation here explaining why this class exists.

**Review by rramsden — APPROVED** [review]
Looks good but the observer module needs more documentation since you had to describe what it does to me on call.

File: src/app/code/Komoju/Payments/Observer/RestoreQuoteFromSession.php (hunk 1 above)
Comment (Dinwy) [reply src/app/code/Komoju/Payments/Observer/RestoreQuoteFromSession.php:24]: Thank you for the feedback! I will add a document for this module. WIP

Comment (Dinwy) [reply src/app/code/Komoju/Payments/Observer/RestoreQuoteFromSession.php:24]: @rramsden I've added [doc block](https://github.com/degica/komoju-magento/pull/14/files#diff-2e6a427ee827e16640c79e37c727b24c2c702bf67c77e536c17fbbfa262f71b6R15) and documentation for [events and observers](https://github.com/degica/komoju-magento/pull/14/files#diff-ab8214b786749cdf3ffa18dcf88f31eccf351ade33354d74f45c322a8fa696bc)
Please check and let me know how you think!

**Pull request merged**